docs/app-rails/application-security.md

```diff
@@ -0,0 +1,136 @@
+# Application Security
+
+Application security is a top priority for government technology application development, which is why the Rails framework's security helper methods and countermeasures help speed up secure application delivery. However, the framework isn't useful by itself; its helper methods and configurations only work if they are used properly. This document uses the Rails Guide to [Securing Rails Applications](https://guides.rubyonrails.org/security.html) to audit the template's use of Rails security best practices. This is meant to be a living document and should be updated as additional security tools and configurations are implemented, as well as when new vulnerabilities are discovered or introduced.
+
+Areas for improvement are marked with `TODO` that describe a security implementation. They should be accompanied by a comment in the codebase referencing the action needed, like `# **TODO:** sanitize input to protect against SQL injection`.
+
+## Conforms to Best Practices
+As outlined in [Securing Rails Applications](https://guides.rubyonrails.org/security.html)
+
+### 1 Introduction
+
+### 2 Sessions
+#### 2.1 What Are Sessions
+#### 2.2 Session Hijacking: 
+1. `config.force_ssl = true` is set for production environments.
+1. Provided the user with a logout button.
+#### 2.3 Session Storage
+1. Using Devise's Session Controller to manage session storage.
+1. Cookies stored client side do not contain sensitive information.
+1. Cookies time out in 15 minutes of inactivity with devise `config.timeout_in`.
+1. Cookies are encrypted client side.
+1. Devise uses BCrypt and the secret_key_base by default for secret hashing.
+1. **TODO:** Cognito, the auth service, is also managing sessions and if wanted to expire sessions after a certain time, regardless of activity, we’d have to do it there.
+#### 2.4 Rotating Encrypted and Signed Cookies Configurations
+1. **TODO:** We are not rotating secrets for hashing nor the hashing algorithm, could do with cognito.
```

Comment:
I think I misunderstood the docs here: https://guides.rubyonrails.org/v7.1/security.html#rotating-encrypted-and-signed-cookies-configurations
I think it's saying that if you need to change your hashing cipher or digest, to use a rotation (via action dispatch) rather than invalidating all existing sessions at once. I'll remove this suggestion.